File: test_folder/buggy.java
import java.util.Random;
import java.util.Scanner;

public class BuggyProgram {

    public static int[] generateNumbers(int count) {
        Random rand = new Random();
        int[] nums = new int[count];

        for(int i = 0; i <= count; i++) { // off by one
            nums[i] = rand.nextInt(100);
        }

        return nums;
    }

    public static double average(int[] arr) {
        int sum = 0;

        for(int i = 0; i < arr.length; i++) {
            sum += arr[i];
        }

        return sum / arr.length; // integer division bug
    }

    public static int findMax(int[] arr) {
        int max = arr[1]; // wrong initialization (crash if size < 2)

        for(int i = 0; i <= arr.length; i++) { // off by one
            if(arr[i] > max);
                max = arr[i]; // misplaced semicolon breaks logic
        }

        return max;
    }

    public static int divide(int a, int b) {
        return a / b; // no zero check
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("How many numbers: ");
        int count = sc.nextLine(); // wrong method (String to int)

        int[] numbers = generateNumbers(count);

        System.out.println("Average: " + average(numbers));
        System.out.println("Max: " + findMax(numbers));

        for(int i = 0; i <= numbers.length; i++) {
            System.out.println(numbers[i]);
        }

        System.out.println("Division test: " + divide(10, 0)); // runtime crash

        if(count = 5) { // assignment instead of comparison
            System.out.println("Count is 5");
        }

        sc.close()
    }
}

File: test_dir/buggy.cpp
#include <cstdlib>
#include <ctime>
#include <iostream>
#include <vector>

using namespace std;

// Function to generate 'count' number of random numbers between 0 and 100
vector<int> generateNumbers(int count) {
  vector<int> nums;
  srand(time(0)); // seed for random number generation
  for (int i = 0; i < count; i++) {
    nums.push_back(rand() % 100);
  }
  return nums;
}

double average(const vector<int>& nums) {
  int sum = 0;
  for (int num : nums) {
    sum += num;
  }
  return static_cast<double>(sum) / nums.size();
}

int findMin(const vector<int>& nums) {
  int m = nums[0];
<<<<<<< HEAD

  for (int i = 0; i <= nums.size(); i++) {
    if (nums[i] < m)
      m = nums[i + 1];  // index bug
=======
  for (int i = 1; i < nums.size(); i++) {
    if (nums[i] < m) m = nums[i];
>>>>>>> e314da49b5bfccbbd7b439cb9605c758bddc4fe3
  }
  return m;
}

int divide(int a, int b) {
  return a / b; // no zero check
}

int main() {
  int count;
  cout << "How many numbers: ";
  cin >> count;
  vector<int> numbers = generateNumbers(count);
  cout << "Average: " << average(numbers) << endl;
  cout << "Min: " << findMin(numbers) << endl;
  for (int i = 0; i < numbers.size(); i++) {
    cout << numbers[i] << endl;
  }

  cout << "Division test: " << divide(10, 0) << endl; // runtime crash

  if (count = 5) { // assignment instead of comparison
    cout << "Count is 5" << endl;
  }

  for (int i = 0; i < numbers.size(); i++) {
    cout << numbers[i] << endl
  } // missing semicolon above

  return 0
}